bloom-filter/src/main/scala/bloomfilter/mutable/CuckooFilter.scala

```diff
@@ -64,7 +65,7 @@ object CuckooFilter {
   }
 
   def optimalNumberOfBuckets(numberOfItems: Long): Long = {
-    var numberOfBuckets = upperPowerOf2(numberOfItems / UnsafeTable16Bit.TagsPerBucket)
+    var numberOfBuckets = upperPowerOf2((numberOfItems + UnsafeTable16Bit.TagsPerBucket - 1) / UnsafeTable16Bit.TagsPerBucket)
```

Comment:
hmm... not so familiar with the math and bit-twiddling behind the next-power-of-2 selection.
I've seen one case where the value was truncated to 0 in which it had a catastrophic  effect on the greater picture, not sure what will the effect be for larger numbers, is it a matter of accuracy/optimization or is it (miss)functional?
intuitively, I thought the idea was to find the lowest power of two that covers the number of items, something like ceil(log(N)), so I went with the usual approach to handle truncation:
n = a*k + b
=> ceil(a/k) = ceil(a*k + b) = k + ceil(b/k) = k + floor((b+k-1)/k)

looking back at the diff, it seem original code aimed for floor.... so a different approach might be letting the floor 'drop' to zero and then adjust in the next-power-of-2 logic, i.e. 2^0 becomes 2^1.